### ISSUE TITLE
v249 breaks running systemd inside a privileged container

### ISSUE THREAD

**Issue by paralin** [systemd/systemd#20707]
After updating from v248 to v249 on a Gentoo container running in Docker on Linux 5.13.15 on x86_64:

```
systemd 249 running in system mode (+PAM -AUDIT -SELINUX -APPARMOR +IMA +SMACK +SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL -ELFUTILS -FID
O2 +IDN2 -IDN -IPTC +KMOD -LIBCRYPTSETUP +LIBFDISK +PCRE2 -PWQUALITY -P11KIT -QRENCODE -BZIP2 +LZ4 -XZ -ZLIB +ZSTD +XKBCOMMON +UTMP +SYSVINIT default-hierarchy=hybrid)                                                                                                                           
Detected virtualization docker.                                                                                                                  
Detected architecture x86-64.                                                                                                                    
                                                                                                                                                 
Welcome to Gentoo/Linux!                                                                                                                         
                                                                                                                                                 
/lib/systemd/system/lightdm.service:8: Standard output type syslog is obsolete, automatically updating to journal. Please update your unit file, 
and consider removing the setting altogether.                                                                                                    
Queued start job for default target Graphical Interface.                                                                                         
Assertion '(_error) != 0' failed at src/core/cgroup.c:766, function cgroup_xattr_apply(). Aborting.                                              
Caught <ABRT>, core dump failed (child 21, code=killed, status=6/ABRT).                                                                          
Exiting PID 1...                                                                                                                                 
```

Downgrading to the previous version v248 fixes the issue.

I assume this has something to do with changes related to cgroups, which must be affected by running in a container. The command to run the container is:

```
docker run --security-opt seccomp=unconfined --name=gentoo -t -d --restart=always --entrypoint=/lib/systemd/systemd --env container=docker --net=host --privileged my/gentoo-container:latest
```

... which works fine with v248.

I've also seen this issue on my Gentoo containers for arm after updating systemd.

**Comment by paralin** [comment]
Here's the working version output:

```
systemd 248 running in system mode. (+PAM -AUDIT -SELINUX -APPARMOR +IMA +SMACK +SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL -ELFUTILS -FIDO2 +IDN2 -IDN -IPTC +KMOD -LIBCRYPTSETUP +LIBFDI
SK +PCRE2 -PWQUALITY -P11KIT -QRENCODE -BZIP2 +LZ4 -XZ -ZLIB +ZSTD +XKBCOMMON +UTMP +SYSVINIT default-hierarchy=hybrid)
Detected virtualization docker.
Detected architecture x86-64.

Welcome to [[1;32mGentoo/Linux!
Queued start job for default target Graphical Interface.
```

**Comment by yuwata** [comment]
Already fixed by 0cddb53c85588fbfb8043f622895c7bd15819198.

**Comment by yuwata** [comment]
And the commit is already backported to v249-stable repository since v249.3. See https://github.com/systemd/systemd-stable/commit/7e79bfce0674c58068d2a125ed666986544e790f.
